fairseq/models/speech_to_speech/s2s_conformer.py

```diff
@@ -57,31 +53,68 @@ def forward(
 @register_model("s2ut_conformer")
 class S2UTConformerModel(S2UTTransformerModel):
     """
-    Direct speech-to-speech translation model with S2T Conformer encoder + Transformer discrete unit decoder
+    Direct speech-to-speech translation model with Conformer encoder + Transformer discrete unit decoder (S2UT)
     """
 
     @staticmethod
     def add_args(parser):
         S2UTTransformerModel.add_args(parser)
-        parser.add_argument("--depthwise-conv-kernel-size", default=31)
+        parser.add_argument("--depthwise-conv-kernel-size", type=int, default=31)
         parser.add_argument(
             "--attn-type",
+            type=str,
             default=None,
             help="If not specified uses fairseq MHA. Other valid option is espnet for using conformer",
         )
         parser.add_argument(
             "--pos-enc-type",
+            type=str,
             default="abs",
             help="Must be specified in addition to attn-type=espnet for rel_pos and rope",
         )
 
     @classmethod
     def build_encoder(cls, args):
-        print(args)
-        data_cfg = S2SDataConfig(Path(args.data) / args.config_yaml)
-        args.input_feat_per_channel = data_cfg.input_feat_per_channel
-        args.input_channels = data_cfg.input_transformed_channels
+        encoder = S2SConformerEncoder(args)
+        pretraining_path = getattr(args, "load_pretrained_encoder_from", None)
+        if pretraining_path is not None:
```

Comment:
Looks like this code is duplicated between TranslatotronConformer and S2UTConformer; is this necessary?

Reply:
@aghatare Thank you for your suggestion. I'll unify these two methods.